saas/backends/stripe_processor/base.py

```diff
@@ -563,3 +586,15 @@ def prorate_transfer(self, amount, provider):
                 # integer division
                 return (amount * 50 + 5000) // 10000
         return 0
+
+    def generate_idempotent_key(self, *data):
+        key = ''
+        if data:
+            hsh = sha512()
+            today = datetime.datetime.utcnow()
```

Comment:
use `saas.utils.datetime_or_now` instead of `datetime.datetime.utcnow()`. This gives an opportunity to replay a timeline.